packages/strapi-connector-bookshelf/lib/migrations/create-migration-runner.js

```diff
@@ -20,8 +23,8 @@ const createMigrationRunner = (migrationFunction, { hooks = [] } = {}) => {
   };
 
   const run = async options => {
-    await beforeHook(options);
-    await migrationFunction(options);
+    const migrationInfos = await beforeHook(options);
```

Comment:
we could create a context to pass to the hooks. so the gooks can set the recreateTable option to true and just interpret this option in the next step